spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPartitionReaderFactory.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.source;
+
+import org.apache.iceberg.ChangelogScanTask;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.read.InputPartition;
+import org.apache.spark.sql.connector.read.PartitionReader;
+import org.apache.spark.sql.connector.read.PartitionReaderFactory;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+
+class SparkPartitionReaderFactory implements PartitionReaderFactory {
+  private final int batchSize;
+
+  SparkPartitionReaderFactory(int batchSize) {
```

Comment:
Maybe we should just doc here that a batch size > 1 will create a ColumnarReader , just seems like a bit of a magic parameter here

Reply:
What about creating separate batch and non-batch reader factories? I just copied the existing code but having separate classes seems more natural than checking if batchSize is > 1.

Thoughts, @RussellSpitzer @szehon-ho?